[Bug]:  Issue with Default session_start_time timezone in mock_NWBFile

### What happened?

### Description

A potential bug occurs when using a mock function for creating `NWBFile` objects. The function set `session_start_time` to  datetime(1970, 1, 1, tzinfo=tzlocal()). This results in an OSError: [Errno 22] Invalid argument when attempting to write the file using pynwb.NWBHDF5IO. The issue seems to be caused by the interaction between tzlocal() and early datetime values (e.g., the epoch time 1970-01-01), particularly in systems located in time zones earlier than UTC, where the local time would fall outside the valid range for the epoch.


### Steps to Reproduce

```python
from pynwb.testing.mock.file import mock_NWBFile

# Create NWBFile object using default session_start_time 
# 
nwbfile = mock_NWBFile()

# Attempt to write the NWB file
with NWBHDF5IO('test.nwb', mode='w') as io:
    io.write(nwbfile)  # This triggers the error
```


### Traceback

```python
...
    return x.isoformat()  # method works for both date and datetime
..…\dateutil\tz\tz.py:291: in _naive_is_dst
    dstval = self._naive_is_dst(dt)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = tzlocal(), dt = datetime.datetime(1970, 1, 1, 0, 0, tzinfo=tzlocal())

    def _naive_is_dst(self, dt):
        timestamp = _datetime_to_timestamp(dt)
>       return time.localtime(timestamp + time.timezone).tm_isdst
E       OSError: [Errno 22] Invalid argument
```


### Operating System

Windows

### Python Executable

Conda

### Python Version

3.11

### Package Versions

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you checked the [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this bug was not already [reported](https://github.com/NeurodataWithoutBorders/pynwb/issues)?

--- a/src/pynwb/testing/mock/file.py
+++ b/src/pynwb/testing/mock/file.py
@@ -10,7 +10,7 @@ from .utils import name_generator
 def mock_NWBFile(
     session_description: str = 'session_description',
     identifier: Optional[str] = None,
-    session_start_time: datetime = datetime(1970, 1, 1, 12, tzinfo=tzutc()),
+    session_start_time: datetime = datetime(1970, 1, 1, tzinfo=tzutc()),
     **kwargs
 ):
     return NWBFile(

--- a/src/pynwb/testing/mock/file.py
+++ b/src/pynwb/testing/mock/file.py
@@ -1,7 +1,7 @@
 from typing import Optional
 from uuid import uuid4
 from datetime import datetime
-from dateutil.tz import tzlocal
+from dateutil.tz import tzutc
 
 from ...file import NWBFile, Subject
 from .utils import name_generator
@@ -10,7 +10,7 @@ from .utils import name_generator
 def mock_NWBFile(
     session_description: str = 'session_description',
     identifier: Optional[str] = None,
-    session_start_time: datetime = datetime(1970, 1, 1, tzinfo=tzlocal()),
+    session_start_time: datetime = datetime(1970, 1, 1, 12, tzinfo=tzutc()),
     **kwargs
 ):
     return NWBFile(

